rustler_mix/lib/rustler/compiler/config.ex

```diff
@@ -32,13 +32,21 @@ defmodule Rustler.Compiler.Config do
 
   alias Rustler.Compiler.Config
 
-  def from(otp_app, module, opts) do
-    config = Application.get_env(otp_app, module, [])
-
+  def from(otp_app, module, config, opts) do
     crate = config[:crate] || opts[:crate] || otp_app
 
     # TODO: Remove in 1.0
-    rustler_crates = Mix.Project.config()[:rustler_crates] || []
+    rustler_crates =
+      if mix_config = Mix.Project.config()[:rustler_crates] do
+        mix_config
+      else
+        IO.warn(
+          ":rustler_crates in mix.exs is deprecated, please explicitly pass options on `use Rustler` or configure the module in your `config/*.exs` files"
+        )
+
+        []
+      end
```

Comment:
@josevalim Am I misunderstanding something or is this logic the wrong way round? We should raise the warning if `:rustler_crates` is found, right?